ocaml-eglot-type-enclosing.el

```diff
@@ -0,0 +1,164 @@
+;;; ocaml-eglot-type-enclosing.el --- Type Enclosing feature   -*- coding: utf-8; lexical-binding: t -*-
+
+;; Copyright (C) 2024-2025  Xavier Van de Woestyne
+;; Licensed under the MIT license.
+
+;; Author: Xavier Van de Woestyne <xaviervdw@gmail.com>
+;; Created: 10 January 2025
+;; SPDX-License-Identifier: MIT
+
+;;; Commentary:
+
+;; Plumbing needed to implement the primitives related to type
+;; enclosing commands.
+
+;;; Code:
+
+(require 'cl-lib)
+(require 'ocaml-eglot-util)
+(require 'ocaml-eglot-req)
+
+;;; Customizable variables
+
+(defcustom ocaml-eglot-type-buffer-name "*ocaml-eglot-types*"
+  "The name of the buffer storing types."
+  :group 'ocaml-eglot
+  :type 'string)
+
+;;; Internal variables
+
+(defvar-local ocaml-eglot-type-enclosing-types nil
+  "Current list of enclosings related to types.")
+
+(defvar-local ocaml-eglot-type-enclosing-current-type nil
+  "Current type for the current enclosing.")
+
+(defvar-local ocaml-eglot-type-enclosing-offset 0
+  "The offset of the requested enclosings.")
+
+(defvar-local ocaml-eglot-type-enclosing-verbosity nil
+  "The verbosity of the current enclosing request.")
```

Comment:
Why not start at 1 ? It would simplify the logic in the increase / decrease verbosity functions...

Reply:
You where right. It is fixed on the last commit!